config/crd/bases/saas.3scale.net_autossls.yaml

```diff
@@ -30,8 +30,9 @@ spec:
         metadata:
           type: object
         spec:
+          description: Spec defines the desired state of AutoSSL
           type: object
-          description: AutoSSLSpec defines the desired state of AutoSSL
+          x-kubernetes-preserve-unknown-fields: true
```

Comment:
Do we want to enable this `x-kubernetes-preserve-unknown-fields: true` on all CRDs?

AFAIK, on a default CRD where there is no kind of validation (fresh operator-sdk operator), it is important to have this value, because you might me defining a spec on your CRs but not validating it at CRD level.

In the moment you define the spec validation on the CRD, I don't think it is a good idea having `x-kubernetes-preserve-unknown-fields: true`, otherwise, any field not specified on the CRD that can be defined on a CR, will pass the api validation and be store on etcd.

https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#field-pruning

What do you think? cc @roivaz

Reply:
After some discussion, we will keep this flag until the operator is more stable.